sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -603,7 +603,14 @@ case class ExternalMapToCatalyst private(
 
   override def foldable: Boolean = false
 
-  override def dataType: MapType = MapType(keyConverter.dataType, valueConverter.dataType)
+  override def dataType: MapType = {
+    val isPrimitiveType = valueType match {
+      case BooleanType | ByteType | ShortType | IntegerType | LongType |
+            FloatType | DoubleType => true
+      case _ => false
+    }
+    MapType(keyConverter.dataType, valueConverter.dataType, !isPrimitiveType)
```

Comment:
I agree with it, but how about this patch (https://github.com/ueshin/apache-spark/commit/1462a767c9643c04b08cb29abaf94f8a5090a90a) for now?
All tests in catalyst/sql passed in my local environment after patch.

I'd like to separate the following issues to other PRs:

- Add `returnNullable: Boolean` to `StaticInvoke` and `Invoke` and fix nullable of them.
  - and then remove `AssertNotNull()` at [here](https://github.com/ueshin/apache-spark/commit/1462a767c9643c04b08cb29abaf94f8a5090a90a#diff-89643554d9757dd3e91abff1cc6096c7R581) like the current implementation.
- Add `valueNullable: Boolean` to `ExternalMapToCatalyst`.
  - and then remove `AssertNotNull()` at [here](https://github.com/ueshin/apache-spark/commit/1462a767c9643c04b08cb29abaf94f8a5090a90a#diff-89643554d9757dd3e91abff1cc6096c7R485) and add `valueNullable = valueType.typeSymbol.asClass.isPrimitive`.

What do you think?